feat(trial balance) display account label on the errors grid

jeremielodi opened the pull request:
closes #3534

jeremielodi committed:
feat(trial balance) display account label on the errors grid

jniles reviewed (commented):
@jeremielodi, this is close.

For each error, we want to say what the error is and how to fix it.  So, basically:
 1. For "Error: locked fiscal year", we want to say "Fiscal year {{fiscalYearLabel}} is locked and must be reopened before data is posted to it."
 2. For "Error: Users is Locked", we want to say "The user {{username}} has a locked account.  Please reopen {{username}} to allow them to post records."
 3. For Unbalanced transactions, we want to say "This transactions has debits equal to {{sumdebits}} and credits equal to {{sumcredits}}.  All transactions must balance before being posted."
... etc.

That is what the goal of https://github.com/IMA-WorldHealth/bhima/issues/3534 was.

jniles commented on server/models/procedures/trial_balance.sql:
```diff
@@ -96,91 +96,107 @@ CREATE PROCEDURE TrialBalanceErrors()
     HAVING COUNT(pj.record_uuid) < 2;
 
-  SELECT DISTINCT BUID(record_uuid) AS record_uuid, trans_id, code FROM stage_trial_balance_errors ORDER BY code, trans_id;
+  SELECT DISTINCT BUID(record_uuid) AS record_uuid, trans_id, account_id, account_label, code FROM stage_trial_balance_errors ORDER BY code, trans_id;
```
Does `account_label` still exist?  It looks like it should be `error_description`.

jniles commented on server/models/procedures/trial_balance.sql:
```diff
@@ -96,91 +96,107 @@ CREATE PROCEDURE TrialBalanceErrors()
   -- check that all periods are unlocked
   INSERT INTO stage_trial_balance_errors
-    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.LOCKED_PERIOD' AS code
+    SELECT pj.record_uuid, MAX(pj.trans_id), ac.id, CONCAT(ac.number,' - ', ac.label)  as error_description, 'POSTING_JOURNAL.ERRORS.LOCKED_PERIOD' AS code
```
Let's just remove `ac.id` from all these.  It looks like we don't use it on the client anyway..

jniles reviewed (commented):
Just a few clean up.  This looks good.

jniles committed:
chore(trial balance): clean up stored procedure

This commit cleans up the trial balance stored procedure and tightens
the error messages.

jniles committed:
deps: update snyk

jniles committed:
Merge remote-tracking branch 'upstream/master' into trialbalance_error
